src/backends/simple.ts

```diff
@@ -0,0 +1,442 @@
+    }
+
+    /**
+     * Get the frame for a given time in seconds
+     * @param targetTime
+     */
+    async getFrameAtTime(targetTime: number): Promise<beamcoder.Frame> {
+        LOG && console.log(`getFrameAtTime time(s)=${targetTime}`);
+        const targetPts = Math.floor(this._timeToPTS(targetTime));
+        return this._getFrameAtPts(targetPts);
+    }
+
+    /**
+     * Get imageData for a given time in seconds
+     * @param targetTime
+     */
+    async getImageDataAtTime(targetTime: number): Promise<ImageData> {
+        const targetPts = Math.floor(this._timeToPTS(targetTime));
+        LOG && console.log('targetTime', targetTime, '-> targetPts', targetPts);
+        const frame = await this._getFrameAtPts(targetPts);
+        if (!frame) {
+            LOG && console.log('no frame found');
+            return null;
+        }
+        const rawData = this._resizeFrameData(frame);
+        const image = createImageData(
+            rawData,
+            frame.width,
+            frame.height
+        ) as ImageData;
+        return image;
+    }
+
+    /**
+     * Get the presentation timestamp (PTS) for a given time in seconds
+     * @param time
+     */
+    _timeToPTS(time: number) {
+        const time_base = this.demuxer.streams[this.streamIndex].time_base;
+        return time * time_base[1] / time_base[0];
+    }
+
+    /**
+     * Get the time in seconds from a given presentation timestamp (PTS)
+     */
+    ptsToTime(pts: number) {
+        const time_base = this.demuxer.streams[this.streamIndex].time_base;
+        return pts * time_base[0] / time_base[1];
+    }
+
+    /**
+     * Get the frame at the given presentation timestamp (PTS)
+     * @param targetPTS
+     */
+    async _getFrameAtPts(targetPTS: number) {
+        LOG && console.log('_getFrameAtPts', targetPTS, '-> duration', this.duration);
+        let packetReadCount = 0;
+
+        // seek and create a decoder when retrieving a frame for the first time or when seeking backwards
+        // we have to create a new decoder when seeking backwards as the decoder currently doesn't support seeking
+        // (it doesn't properly release frames when seeking backwards)
+        if (!this.previousTargetPTS || this.previousTargetPTS > targetPTS) {
+            await this.demuxer.seek({
+                stream_index: 0, // even though we specify the stream index, it still seeks all streams
+                timestamp: targetPTS,
+                any: false,
+            });
+            this._createDecoder();
+            this.packet = null;
+            this.frames = [];
+        }
+
+        // the decoder has been previously flushed while retrieving frames at the end of the stream and has thus been
+        // destroyed. See if the requested targetPTS is part of the last few frames we decoded. If so, return it.
+        if (!this.decoder) {
+            LOG && console.log('no decoder');
+            if ((this.filteredFrames as any).length > 0) {
```

Comment:
```suggestion
            if (this.filteredFrames.length > 0) {
```

Reply:
It's because filtered frames can be either `undefined` or an array

Reply:
Ahh, so the code would be clearer like this? does that compile?

```suggestion
            if (this.filteredFrames && this.filteredFrames.length > 0) {
```

Reply:
That doesn't work @antoineMoPa